test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/AuthorizationApplicationModelProviderTest.cs

```diff
@@ -81,6 +81,29 @@ public void CreateControllerModelAndActionModel_AllowAnonymousAttributeAddsAllow
             Assert.Single(action.Filters, f => f is AllowAnonymousFilter);
         }
 
+        [Fact]
+        public void CreateControllerModelAndActionModel_NoAuthNoFilter()
+        {
+            // Arrange
+            var provider = new AuthorizationApplicationModelProvider(
+                new DefaultAuthorizationPolicyProvider(
+                    new TestOptionsManager<AuthorizationOptions>()
+                ));
+            var defaultProvider = new DefaultApplicationModelProvider(new TestOptionsManager<MvcOptions>());
+
+            var context = new ApplicationModelProviderContext(new[] { typeof(NoAuthController).GetTypeInfo() });
+            defaultProvider.OnProvidersExecuting(context);
+
+            // Act
+            provider.OnProvidersExecuting(context);
+
+            // Assert
+            var controller = Assert.Single(context.Result.Controllers);
+            Assert.Empty(controller.Filters);
+            var action = Assert.Single(controller.Actions);
+            Assert.Empty(action.Filters);
+        }
```

Comment:
We should probably add a test with a policy provider that does return different requirements for a policy each time(maybe just return +1 requirements each time its called), to ensure that scenario still works